### PR TITLE
gh: reorg on ci files

### PR DESCRIPTION
**Pull request by beraldoleal**
Lets move specific architecture workflows to their own folders for better organization.

If this is well-received, more changes could follow.

### CODE REVIEW

**Comment by beraldoleal** [comment]
@fidencio @wainersm @gkurz let me know if this type of reorg is welcome and makes sense.

**Review by c3d — COMMENTED** [review]
For what it's worth, I like it @beraldoleal

File: .github/workflows/amd64/basic-ci.yaml
Comment (gkurz) [inline .github/workflows/amd64/basic-ci.yaml:1]: Hi @beraldoleal !

First, I definitely like the idea of structuring these files a bit. It seems right to move all the files with an `${arch}` suffix to an `${arch}` subdir...

File: .github/workflows/amd64/run-k8s-tests-on-aks.yaml
Comment (gkurz) [inline .github/workflows/amd64/run-k8s-tests-on-aks.yaml:1]: This one doesn't have an architecture suffix but it requires the amd64 static tarball :
```
.github/workflows/run-k8s-tests-on-aks.yaml:          name: kata-static-tarball-amd64${{ inputs.tarball-suffix }}
```
so maybe it is acceptable to move it under the `amd64`.

File: .github/workflows/amd64/run-k8s-tests-on-garm.yaml
Comment (gkurz) [inline .github/workflows/amd64/run-k8s-tests-on-garm.yaml:1]: This one doesn't seem to be arch specific. Why moving it under amd64 ?

Comment (gkurz) [reply .github/workflows/amd64/run-k8s-tests-on-garm.yaml:1]: Same question for all the other ones without an architecture suffix actually.

Comment (beraldoleal) [reply .github/workflows/amd64/run-k8s-tests-on-garm.yaml:1]: I moved because is depending on amd64 binaries.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -126,7 +126,7 @@ jobs:
 
   run-kata-deploy-tests-on-garm:
     needs: publish-kata-deploy-payload-amd64
-    uses: ./.github/workflows/run-kata-deploy-tests-on-garm.yaml
+    uses: ./.github/workflows/amd64/run-kata-deploy-tests-on-garm.yaml
```
Comment (beraldoleal) [inline .github/workflows/ci.yaml:129]: @gkurz here, see line 128. needs amd64 payload.

**Comment by wainersm** [comment]
Hi @beraldoleal ! Nice idea but... :(

... last time I tried to organize action files in directories I realized github expect all them flat in `./.github/workflows`. I even found an issue with people suggesting allowing to have hierarchies of directories. Have it changed?

**Comment by wainersm** [comment]
> Hi @beraldoleal ! Nice idea but... :(
> 
> ... last time I tried to organize action files in directories I realized github expect all them flat in `./.github/workflows`. I even found an issue with people suggesting allowing to have hierarchies of directories. Have it changed?

I just found this: https://github.com/orgs/community/discussions/18055

**Comment by beraldoleal** [comment]
> > Hi @beraldoleal ! Nice idea but... :(
> > ... last time I tried to organize action files in directories I realized github expect all them flat in `./.github/workflows`. I even found an issue with people suggesting allowing to have hierarchies of directories. Have it changed?
> 
> I just found this: [github.com/orgs/community/discussions/18055](https://github.com/orgs/community/discussions/18055)

Yep, just realized that! :( `.github/actions` is reserved for something else and cannot be used either. I had hopes we could use outside `.github/workflows`. We would need to convert that to actions or services. :(

**Pull request closed without merging**